### ISSUE TITLE
Designate some journals as archived

### ISSUE THREAD

**Issue by joemull** [openlibhums/janeway#3970]
**Is your feature request related to a problem? Please describe.**
As a press manager, I'd like to designate some journals as archived, so that I can list them separately from the actively publishing journals on the [journal list page](https://www.openlibhums.org/journals/).

This was requested by Caroline during the Hourglass theme launch.

**Describe the solution you'd like**
I think we can handle this and future journal status functionality with a new "status" field on journals that has a dropdown / enumeration for the different states a journal can be in, such as:

* hidden
* coming_soon
* active
* archived

We could then change `journal.hide_from_press` to an object property that uses status.

Going forward, we can use properties like `public_journals` on the press to get the list of journals we want.

**Describe alternatives you've considered**
A Django object manager, but that might be overkill and confusing for something that is more properly an enumeration.

**Comment by joemull** [comment]
Bringing over a comment @ajrbyers made on another issue: "The options may need to be a bit more granular. A journal can be active (e.g. publishing) and also hidden from the press (and thus should not be included in the press OAI)."

Maybe `hidden_from_press` should really just be a visibility boolean, while `status` should be an enumeration that can be used to create subgroups of visible journals. We'd have to expand the current `public_journals` with a few more properties like `public_active_journals` and `public_archived_journals` to get the querysets we want in templates, but it would probably work ok.

**Comment by joemull** [comment]
Closed by mistake

**Comment by S-Haime** [comment]
By the looks of it, this is being actively worked on. If so, feel free to remove the parent issue :)

**Comment by joemull** [comment]
Yes at this point it's just waiting on another review from Andy, so I'll remove the parent issue.
